src/components/List.js

```diff
@@ -92,11 +74,9 @@ const List = ({ token }) => {
     /* Added 1 here since we're working with full days
      * if user taps a checkbox and the timeFrame was 7 days, 6 would have been displayed
      */
-    const daysUntilNextPurchase = differenceInDays(
-      nextPurchasedDate,
-      Date.now(),
-    );
-    return daysUntilNextPurchase > 0 ? daysUntilNextPurchase + 1 : 0;
+    const daysUntilNextPurchase =
+      differenceInDays(nextPurchasedDate, Date.now()) + 1;
+    return daysUntilNextPurchase > 0 ? daysUntilNextPurchase : 0;
```

Comment:
Is this always greater than zero because you're adding 1 on line 78?

Reply:
I believe that this is what @veeteeran was talking about where if, for example, the value we get from `differenceInDays(nextPurchasedDate, Date.now( ))` should be 7 days, because it automatically starts counting down, the second we have even a millisecond less than 7 it rounds it all the way down to 6 days. So to counter that almost instantaneous action of rounding down, we add 1 to the number we get from `differenceInDays(nextPurchasedDate, Date.now( ))`